merkledag/node.go

```diff
@@ -83,6 +84,26 @@ func (l *Link) GetNode(ctx context.Context, serv DAGService) (*Node, error) {
 	return serv.Get(ctx, key.Key(l.Hash))
 }
 
+// GetNodeAndCache return the MDAG Node that the link points to and store a
+// pointer to that node along with the link to speed up further retrivals. A
+// timeout is to be specified to avoid taking too much time.
+func (l *Link) GetNodeAndCache(ctx context.Context, serv DAGService, timeout time.Duration) (*Node, error) {
```

Comment:
timeout should be specified on the context using `context.WithTimeout` on the caller.